File: 02_Estructuras de Control/02_Condicionales/03_Anidadas/CSharp/ejemplo2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ejemplo2Anidado
{
    internal class Program
    {
        static void Main(string[] args)
        {
            /* Programa para Determinar el Estado Académico de un Alumno con Indicadores Visuales por Rango de Desempeño
             */

            Console.ForegroundColor = ConsoleColor.DarkCyan;
            Console.WriteLine("===== DETERMINACIÓN DEL ESTADO ACADÉMICO DEL ALUMNO =====");
            Console.ResetColor();
            Console.WriteLine();

            // Entrada datos

            Console.Write("Ingrese la primera nota: ");
            double nota1 = double.Parse(Console.ReadLine());

            Console.Write("Ingrese la segunda nota: ");
            double nota2 = double.Parse(Console.ReadLine());

            Console.Write("Ingrese la tercera nota: ");
            double nota3 = double.Parse(Console.ReadLine());

            // Proceso

            double promedio = (nota1 + nota2 + nota3) / 3;

            string estado;

            if (promedio >= 10.0)
            {
                estado = "Fuera de rango";
            }
            else if (promedio >= 7.0)
            {
                estado = "Promocionado";
            }
            else if (promedio >= 4.0)
            {
                estado = "Regular";
            }
            else
            {
                estado = "Reprobado";
            }

            // Salida de resultados

            Console.WriteLine();
            Console.WriteLine($"Resultados:");
            Console.WriteLine($"Nota 1: {nota1}");
            Console.WriteLine($"Nota 2: {nota2}");
            Console.WriteLine($"Nota 3: {nota3}");
            Console.WriteLine();
            Console.WriteLine($"Promedio: {promedio:F2}");
            Console.WriteLine();
            Console.WriteLine($"Estado académico: {estado}");

            Console.WriteLine("\nPresione cualquier tecla para salir...");
            Console.ReadKey();
        }
    }
}

File: 02_Estructuras de Control/02_Condicionales/03_Anidadas/Python/ejemplo2.py
# Programa para Determinar el Estado Académico de un Alumno según su Promedio

print("===== DETERMINACIÓN DEL ESTADO ACADÉMICO DEL ALUMNO =====")
print()

# Entrada de datos
try:
    nota1 = float(input("Ingrese la primera nota: "))
    nota2 = float(input("Ingrese la segunda nota: "))
    nota3 = float(input("Ingrese la tercera nota: "))
except ValueError:
    print("Error: Debe ingresar valores numéricos.")
    exit()

# Proceso: cálculo del promedio
promedio = (nota1 + nota2 + nota3) / 3

# Determinación del estado académico (estructura condicional anidada en cascada)
if promedio >= 10.0:
    estado = "Fuera de rango"
elif promedio >= 7.0:
    estado = "Promocionado"
elif promedio >= 4.0:
    estado = "Regular"
else:
    estado = "Reprobado"

# Salida de resultados
print()
print("Resultados:")
print(f"Nota 1: {nota1}")
print(f"Nota 2: {nota2}")
print(f"Nota 3: {nota3}")
print()
print(f"Promedio: {promedio:.2f}")
print()
print(f"Estado académico: {estado}")